--- thunderhub/arm32v7.Dockerfile	2021-06-22 23:50:12.955684258 +0100
+++ Dockerfile.temp	2021-06-29 01:16:18.495412352 +0100
@@ -1,22 +1,27 @@
 # ---------------
 # Install Dependencies
 # ---------------
-FROM arm32v7/node:14.15-alpine as deps
+FROM arm32v7/node:16-alpine3.11 as deps
 
 WORKDIR /app
 
 # Install dependencies neccesary for node-gyp on node alpine
 RUN apk add --update --no-cache \
     libc6-compat \
-    python \
+    python3 \
     make \
     g++
 
 
 # Install app dependencies
-COPY package.json package-lock.json ./
+COPY ./thunderhub/package.json ./thunderhub/package-lock.json ./
 RUN npm install
 
+FROM arm32v7/golang:1.14.15-alpine3.11 as yqbuild
+
+ENV GO111MODULE=on
+RUN go get github.com/mikefarah/yq/v4
+
 # ---------------
 # Build App
 # ---------------
@@ -30,7 +35,7 @@
 ENV NEXT_TELEMETRY_DISABLED=1
 
 # Build the NextJS application
-COPY . .
+COPY ./thunderhub .
 RUN npm run build
 
 # Remove non production necessary modules
@@ -39,7 +44,10 @@
 # ---------------
 # Release App
 # ---------------
-FROM arm32v7/node:14.15-alpine
+FROM arm32v7/node:16-alpine3.11
+
+RUN echo https://dl-cdn.alpinelinux.org/alpine/edge/community >> /etc/apk/repositories
+RUN apk add --update --no-cache yq bash coreutils
 
 WORKDIR /app
 
@@ -52,9 +60,12 @@
 COPY --from=build /app/public ./public
 COPY --from=build /app/node_modules/ ./node_modules 
 COPY --from=build /app/.next/ ./.next
+COPY --from=yqbuild /go/bin/yq /usr/bin/yq
 
-COPY ./scripts/initCookie.sh ./scripts/initCookie.sh
+COPY ./thunderhub/scripts/initCookie.sh ./scripts/initCookie.sh
 
 EXPOSE 3000 
 
-CMD [ "npm", "start" ]
\ No newline at end of file
+COPY ./docker_entrypoint.sh /usr/local/bin/docker_entrypoint.sh
+
+ENTRYPOINT ["/usr/local/bin/docker_entrypoint.sh"]
